File: python/euler_3.py
# What is the largest prime factor of the number 600851475143?

import math


def is_prime(n):

    for i in range(3, int(math.sqrt(n))):
        if i in (0, 1):
            pass
        elif not n % i:
            return False
    else:
        return True


def factor_factory(n):

    for i in range(3, int(math.sqrt(n) + 1), 2 if n % 2 else 1):
        if n % i == 0:
            yield i


def main(n):

    gpf = 0

    for num in factor_factory(n):
        if is_prime(num):
            gpf = num
        else:
            pass

    return gpf


print('gpf:', main(600851475143))

File: c/euler3.c
#include <stdio.h>
#include <math.h>


int is_prime(int num)
{
	int i = 0;
	
	for(i = 2; i < num; i++) {
		if(num % i == 0) {
			return 1;
		}
	}
	
	return 0;
}



int main()
{
	long limit = 13195;
	int n = 2;
	
	for(n = 2; n < limit; n++) {
		if(limit % n == 0) {
			if(is_prime(n) == 0) {
				printf("%d\n", n);
			}
		}
	}

	return 0;
}	
